abbyy_to_epub3/parse_abbyy.py

```diff
@@ -203,286 +269,239 @@ def parse_abbyy(self):
-                        }
+                # Skip headers and footers
+                if role == 'rt':
+                    continue
 
-                        # To help with unmarked header recognition
-                        if newpage:
-                            d['first'] = True
-                            newpage = False
+                # This is a good text chunk. Instantiate the block.
+                # The modern ABBYY parser is consistent in its handling
+                # of EOL hyphens, making it safe to strip them.
+                d = {
+                    'type': 'Text',
+                    'page_no': self.page_no,
+                    'text': sanitize_xml(text).replace('¬\n', ''),
+                    'role': role,
+                    'style': self.paragraphs[para_id]
+                }
 
-                        # Mark up heading level
-                        if role == 'heading':
-                            level = self.paragraphs[para_id]['roleLevel']
-                            # shortcut so we need fewer lookups later
-                            d['heading'] = level
+                # To help with unmarked header recognition
+                if self.newpage:
+                    d['first'] = True
+                    self.newpage = False
 
-                        # Whenever you append to the list, re-instantiate
-                        self.blocks.append(d)
-                        d = dict()
+                # Mark up heading level
+                if role == 'heading':
+                    level = self.paragraphs[para_id]['roleLevel']
+                    # shortcut so we need fewer lookups later
+                    d['heading'] = level
 
-                        para.clear()  # garbage collection
-                    del paras         # garbage collection
+                # Whenever you append to the list, re-instantiate
+                self.blocks.append(d)
+                d = dict()
 
-                elif self.is_block_type(block, "Table"):
-                    # We'll process the table by treating each of its cells
-                    # subordinate blocks as separate. Keep track of which
-                    # is the last element in a cell/row/table, so we can
-                    # close the elements after each is complete.
-                    this_row = 1
+                para.clear()  # garbage collection
+            del paras         # garbage collection
+
+        elif self.is_block_type(blockattr, "Table"):
+            # We'll process the table by treating each of its cells'
+            # subordinate blocks as separate. Keep track of which  is the last
+            # element in a cell/row/table, so we can close the elements after
+            # each is complete.
+            this_row = 1
+            d = {
+                'type': 'Table',
+                'style': blockattr,
+                'page_no': self.page_no,
+            }
+            self.blocks.append(d)
```

Comment:
Same Q as above: Why not just append the data structure (rather than assigning the structure to a variable and then explicitly clearing it after)? Is it to ensure variables in the dict are out of scope and garbage collected?